File: EASSET.Web/Modules/Perencanaan/RKBMD/RKBMDForm.cs
using EASSET.MasterData;
using Serenity.ComponentModel;
using System;

namespace EASSET.Perencanaan.Forms;

[FormScript("Perencanaan.RKBMD")]
[BasedOnRow(typeof(RKBMDRow), CheckNames = true)]
public class RKBMDForm
{
    //public string UnitId { get; set; }
    //public double NoUrut { get; set; }
    //public string Keterangan { get; set; }
    //public string ThAng { get; set; }
    //public string Op { get; set; }
    //public DateTime Lu { get; set; }
    //public string Pc { get; set; }
    //public short Dlt { get; set; }
    //public string SubUnitId { get; set; }
    public string KodeRekening { get; set; }
    public string NamaRekening { get; set; }
    [DisplayName("Jumlah Usulan")]
    public double JmlUsulan { get; set; }
    public double JmlKebMax { get; set; }
    public double JmlExisting { get; set; }
    public string SubKegNama { get; set; }
    [TextAreaEditor(Rows = 5, Cols = 10)]
    public string Keterangan { get; set; }
    [Hidden]
    public string SubSubRo108Id { get; set; }
    [Hidden]
    public string SubKegId { get; set; }
    [Hidden]
    public string SubKegKode { get; set; }
    //public string Satuan { get; set; }
    //public string Rkpdid { get; set; }
}

File: EASSET.Web/Modules/ServerTypes/Perencanaan/RKBMDForm.ts
import { PrefixedContext, initFormType } from "@serenity-is/corelib";
import { StringEditor, DecimalEditor, TextAreaEditor } from "serenity.corelib";

export interface RKBMDForm {
    KodeRekening: StringEditor;
    NamaRekening: StringEditor;
    JmlUsulan: DecimalEditor;
    JmlKebMax: DecimalEditor;
    JmlExisting: DecimalEditor;
    SubKegNama: StringEditor;
    Keterangan: TextAreaEditor;
    SubSubRo108Id: StringEditor;
    SubKegId: StringEditor;
    SubKegKode: StringEditor;
}

export class RKBMDForm extends PrefixedContext {
    static readonly formKey = 'Perencanaan.RKBMD';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!RKBMDForm.init)  {
            RKBMDForm.init = true;

            var w0 = StringEditor;
            var w1 = DecimalEditor;
            var w2 = TextAreaEditor;

            initFormType(RKBMDForm, [
                'KodeRekening', w0,
                'NamaRekening', w0,
                'JmlUsulan', w1,
                'JmlKebMax', w1,
                'JmlExisting', w1,
                'SubKegNama', w0,
                'Keterangan', w2,
                'SubSubRo108Id', w0,
                'SubKegId', w0,
                'SubKegKode', w0
            ]);
        }
    }
}